hal/ia32/init.h

```diff
@@ -82,13 +118,16 @@ typedef struct {
 	u8 entries[]; /* It is an array of variable length elements */
 } __attribute__ ((packed)) madt_header_t;
 
+
 typedef struct {
-	u8 addressSpaceId;
-	u8 registerWidth;
-	u8 registerOffset;
-	u8 accessSize;
-	u64 address;
-} __attribute__ ((packed)) generic_address_structure_t;
+	sdt_header_t header;
+	u32 eventTimerBlockID;
+	generic_address_structure_t baseAddress;
+	u8 hpetNumber;
+	u16 minPeriodicClockTick;
+	u8 pageProtection;
+} __attribute__ ((packed)) hpet_header_t;
```

Comment:
All structs in hal should have hal_ prefix 
```suggestion
} __attribute__ ((packed)) hal_hpetHeader_t;
```